scripts/binary_to_array.py

```diff
@@ -56,6 +56,13 @@
                         help="Output source file, defining the array data.")
 arg_parser.add_argument("--output_header",
                         help="Output header file, declaring the array data.")
+arg_parser.add_argument("--logical_directory",
```

Comment:
This is my attempt to make sure the script stays generic and is complaint with our linter. Do you think it's overkill?

Reply:
Done, thanks. I really like the idea of using `git rev-parse` here.

I have originally wanted to also exclude the file from linter and `style.sh` because checking it caused a noticeable slowdown. However, now that I think about it, the generated files should only be checked if they are modified, so this shouldn't affect the regular development; therefore, I did not exclude them.